java/arcs/core/data/SchemaFields.kt

```diff
@@ -11,8 +11,40 @@
 
 package arcs.core.data
 
+/** The possible types for a field in a [Schema]. */
+sealed class FieldType(
+    val tag: Tag
+) {
+    /** An Arcs primitive type. */
+    data class Primitive(val primitiveType: PrimitiveType): FieldType(Tag.Primitive)
+
+    /** A reference to an entity. */
+    data class EntityRef(val schemaHash: String) : FieldType(Tag.EntityRef)
+
+    // TODO: Collections of References in fields are not supported in Kotlin at all yet.
```

Comment:
They aren't? You should be able to make a collection of anything referencable, including a dumb reference instance which only has a storage key.

Reply:
I was thinking more at the public sdk level. You're right though, the storage stack should support it. I think I can just delete this comment, it's actually not needed here. The knowledge of whether the field is a singleton or collection is kept elsewhere (in the SchemaFields singleton and collection fields).